src/containerapp/azext_containerapp/custom.py

```diff
@@ -2638,3 +2642,87 @@ def list_environment_telemetry_otlp(cmd,
+    raw_parameters = locals()
+    dotnet_component_decorator = DotNetComponentDecorator(
+        cmd=cmd,
+        client=DotNetComponentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    return dotnet_component_decorator.list()
+
+
+def show_dotnet_component(cmd, dotnet_component_name, environment_name, resource_group_name):
+    raw_parameters = locals()
+    dotnet_component_decorator = DotNetComponentDecorator(
+        cmd=cmd,
+        client=DotNetComponentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    result = dotnet_component_decorator.show()
+    if result is not None:
+        logger.warning("Found DotNet Component '%s' in environment '%s' in resource group '%s'.", dotnet_component_name, environment_name, resource_group_name)
+
+    aspire_dashboard_url = _get_aspire_dashboard_url(cmd, environment_name, resource_group_name, dotnet_component_name)
+    component_type = safe_get(result, "properties", "componentType")
+    if component_type == DOTNET_COMPONENT_RESOURCE_TYPE:
+        logger.warning("Aspire Dashboard URL: %s.", aspire_dashboard_url)
+    return result
+
+
+def delete_dotnet_component(cmd, dotnet_component_name, environment_name, resource_group_name, no_wait):
+    exising_dotnet_component = _get_dotnet_component_if_exists(cmd, dotnet_component_name, environment_name, resource_group_name)
+    if not exising_dotnet_component:
+        raise ValidationError("DotNet Component '{}' does not exist in environment '{}' in resource group '{}'.".format(dotnet_component_name, environment_name, resource_group_name))
+    raw_parameters = locals()
+    dotnet_component_decorator = DotNetComponentDecorator(
+        cmd=cmd,
+        client=DotNetComponentPreviewClient,
+        raw_parameters=raw_parameters,
+        models=CONTAINER_APPS_SDK_MODELS
+    )
+    logger.warning("Deleting DotNet Component '%s' in environment '%s' in resource group '%s'.", dotnet_component_name, environment_name, resource_group_name)
+    result = dotnet_component_decorator.delete()
+    if result is None:
+        logger.warning("Successfully deleted DotNet Component '%s' in environment '%s' in resource group '%s'.", dotnet_component_name, environment_name, resource_group_name)
+    return result
+
+
+def create_dotnet_component(cmd, dotnet_component_name, environment_name, resource_group_name, no_wait, dotnet_component_type=DOTNET_COMPONENT_RESOURCE_TYPE):
+    # Check if DotNet component already exists in environment
+    existing_dotnet_component = _get_dotnet_component_if_exists(cmd, dotnet_component_name, environment_name, resource_group_name)
+    if existing_dotnet_component:
+        raise ValidationError("DotNet Component '{}' already exists in environment '{}' in resource group '{}'. Please delete the DotNet Component before creating a new one.".format(dotnet_component_name, environment_name, resource_group_name))
```

Comment:
Recommend moving this validation logic to `DotNetComponentCreateDecorator.validate_arguments()`

Reply:
Updated.